src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/visits/VisitsMigrationIntTest.kt

```diff
@@ -159,19 +165,25 @@ class VisitsMigrationIntTest : SqsIntegrationTestBase() {
         )
       }
 
-      webTestClient.get().uri("/history")
-        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_VISITS")))
-        .header("Content-Type", "application/json")
-        .exchange()
-        .expectStatus().isOk
-        .expectBody()
-        .jsonPath("$.size()").isEqualTo(1)
-        .jsonPath("$[0].migrationId").isNotEmpty
-        .jsonPath("$[0].whenStarted").isNotEmpty
-        .jsonPath("$[0].whenEnded").isNotEmpty
-        .jsonPath("$[0].estimatedRecordCount").isEqualTo(26)
-        .jsonPath("$[0].migrationType").isEqualTo("VISITS")
-        .jsonPath("$[0].status").isEqualTo("COMPLETED")
+      await untilAsserted {
```

Comment:
wrap the whole thing in a retry else occasionally fails since migration is in the processing of ending, but hasn't quite  written the row yet